src/services/ledger/index.ts

```diff
@@ -60,30 +63,34 @@ export const LedgerService = (): ILedgerService => {
   const getTransactionById = async (
     id: LedgerTransactionId,
   ): Promise<LedgerTransaction | LedgerServiceError> => {
+    let txnRecord: TransactionRecord
     try {
       const _id = toObjectId<LedgerTransactionId>(id)
       const { results } = await MainBook.ledger({
         account_path: liabilitiesMainAccount,
         _id,
       })
-      if (results.length === 1) {
-        return translateToLedgerTx(results[0])
+      txnRecord = results[0]
+      if (results.length !== 1) {
+        return new CouldNotFindTransactionError()
       }
-      return new CouldNotFindTransactionError()
     } catch (err) {
       return new UnknownLedgerError(err)
     }
+
+    return translateTxRecordsToLedgerTxs([txnRecord])[0]
```

Comment:
are you sure this cant throw an  exception? I guess is better to include it in the try block

Reply:
Fixed